besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -604,26 +603,43 @@ private WorldStateArchive createWorldStateArchive(
   protected List<PeerValidator> createPeerValidators(final ProtocolSchedule protocolSchedule) {
     final List<PeerValidator> validators = new ArrayList<>();
 
-    final OptionalLong daoBlock = configOptionsSupplier.get().getDaoForkBlock();
-    if (daoBlock.isPresent()) {
-      // Setup dao validator
-      validators.add(
-          new DaoForkPeerValidator(protocolSchedule, metricsSystem, daoBlock.getAsLong()));
-    }
+    configOptionsSupplier
+        .get()
+        .getDaoForkBlock()
+        .ifPresent(
+            daoBlock ->
+                validators.add(
+                    new DaoForkPeerValidator(protocolSchedule, metricsSystem, daoBlock)));
 
-    final OptionalLong classicBlock = configOptionsSupplier.get().getClassicForkBlock();
-    // setup classic validator
-    if (classicBlock.isPresent()) {
-      validators.add(
-          new ClassicForkPeerValidator(protocolSchedule, metricsSystem, classicBlock.getAsLong()));
-    }
+    configOptionsSupplier
+        .get()
+        .getClassicForkBlock()
+        .ifPresent(
+            classicBlock ->
+                validators.add(
+                    new ClassicForkPeerValidator(protocolSchedule, metricsSystem, classicBlock)));
 
     for (final Map.Entry<Long, Hash> requiredBlock : requiredBlocks.entrySet()) {
       validators.add(
           new RequiredBlocksPeerValidator(
               protocolSchedule, metricsSystem, requiredBlock.getKey(), requiredBlock.getValue()));
     }
 
+    configOptionsSupplier
```

Comment:
This is not as readable as a couple of `if` blocks.  Two if blocks would make it obvious we are adding a config if two config items are persent.  get()\.get<foo>().filter(...).ifPresent(...) is much too verbose, especially when one of the ifPresent paths is nested.

Reply:
done